src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs

```diff
@@ -112,6 +127,25 @@ private void EndRequest(HttpContext httpContext, long timestamp)
                     telemetry.Name = httpContext.Request.Method + " " + httpContext.Request.Path.Value;
                 }
 
+                if (string.IsNullOrEmpty(telemetry.Context.InstrumentationKey))
```

Comment:
Source is only a field on request telemetry (http calls). It is not a custom field. The only thing that it is used for right now is to contain the cross component correlation id (hashed ikey of the callee). Exceptions and events etc. may or may not be in the context of an http call. Also request failures are also captured by request telemetry with a failure status.

The intent so far is to capture data to drive experiences dealing with cross component communication, hence concentrating on request and dependency telemetry types. If we do in future come up with experiences where other telemetry gathered while processing the request needs such information - it can be added at that time by updating the backend schema and adding collection logic in the sdk.